src/seam-connect/routes.ts

```diff
@@ -388,7 +389,11 @@ export abstract class Routes {
     update: async (
       params: AccessCodeUpdateRequest,
       options?: { waitForCompletion?: boolean }
-    ): Promise<ActionAttempt<"UPDATE_ACCESS_CODE">> => {
+    ): Promise<
+      typeof options extends { waitForCompletion: true }
+        ? AccessCode
+        : ActionAttempt<"UPDATE_ACCESS_CODE">
+    > => {
```

Comment:
I think in any case this function returns an action attempt, if waitForCompletion=true then it returns a successful action attempt with a result

Reply:
Oh yep, you are right. Was working on an older branch, my bad!